### PR TITLE
replace adapters by File objects to be more flexible

### PR DESCRIPTION
**Pull request by jaapio**
This pr is depending on https://github.com/phpDocumentor/ReflectionCommon/pull/6 therefor a new tag should be created. And the component should be updated in the composer.json/lock

### CODE REVIEW

File: src/phpDocumentor/Reflection/Php/Factory/File.php [hunk 1]
```diff
@@ -78,12 +71,12 @@ public function __construct(NodesFactory $nodesFactory, Adapter $adapter = null,
...
      *
-     * @param string $filePath path to check.
+     * @param string $file path to check.
      * @return boolean
      */
-    public function matches($filePath)
+    public function matches($file)
     {
-        return is_string($filePath) && $this->adapter->fileExists($filePath);
+        return $file instanceof FileSystemFile;
```
Comment (mvriel) [inline src/phpDocumentor/Reflection/Php/Factory/File.php]: Shouldn't this test the interface and not the concrete instance?

File: src/phpDocumentor/Reflection/Php/Factory/File.php [hunk 2]
```diff
@@ -91,7 +84,7 @@ public function matches($filePath)
      * Since an object might contain other objects that need to be converted the $factory is passed so it can be
      * used to create nested Elements.
      *
-     * @param string $object path to the file to convert to an File object.
+     * @param FileSystemFile $object path to the file to convert to an File object.
```
Comment (mvriel) [inline src/phpDocumentor/Reflection/Php/Factory/File.php]: Shouldn't this be the interface and not the concrete instance?

File: src/phpDocumentor/Reflection/Php/Factory/File.php (hunk 1 above)
Comment (jaapio) [inline src/phpDocumentor/Reflection/Php/Factory/File.php]: It is the interface, `use phpDocumentor\Reflection\File as FileSystemFile;`

File: src/phpDocumentor/Reflection/Php/Factory/File/CreateCommand.php [hunk 3]
```diff
@@ -38,36 +34,23 @@
...
-    {
-        return $this->adapter;
-    }
-
-    /**
      * Returns the path of the processed file.
      *
      * @return string
      */
     public function getFilePath()
```
Comment (mvriel) [inline src/phpDocumentor/Reflection/Php/Factory/File/CreateCommand.php]: the term Path here might be confusing

File: src/phpDocumentor/Reflection/Php/Factory/File.php (hunk 1 above)
Comment (mvriel) [inline src/phpDocumentor/Reflection/Php/Factory/File.php]: Ah, sorry :)

File: src/phpDocumentor/Reflection/Php/Factory/File.php (hunk 2 above)
Comment (mvriel) [inline src/phpDocumentor/Reflection/Php/Factory/File.php]: Never mind :)

**Comment by mvriel** [comment]
Missing tests for LocalFile object and CreateCommand?

**Comment by jaapio** [comment]
tests are added now

**Pull request merged**